extra_roles/roles.js

```diff
@@ -0,0 +1,263 @@
+
+(function(g, e, l){
```

Comment:
How do you expect people to maintain this contribution if all your variables/functions are single letters, devoid of meaning? I'm trying to code review what this change does, and find myself struggling to make any sense of it.